src/mqtt_client.rs

```diff
@@ -500,35 +490,38 @@ where
                 }
 
                 Command::Get { path } => {
-                    let mut data = [0u8; MESSAGE_SIZE];
-                    match self.settings.get_json(path, &mut data) {
-                        Err(err) => err.into(),
-                        Ok(len) => {
-                            let mut topic = self.prefix.clone();
-
-                            // Note(unwrap): We check that the string will fit during
-                            // construction.
-                            topic
-                                .push_str("/settings")
-                                .and_then(|_| topic.push_str(path))
-                                .unwrap();
-
-                            // Note(unwrap): This construction cannot fail because there's always a
-                            // valid topic.
-                            let message = minimq::Publication::new(&data[..len])
-                                .reply(properties)
-                                // Override the response topic with the path.
-                                .topic(&topic)
-                                .qos(QoS::AtLeastOnce)
-                                .finish()
-                                .unwrap();
-
-                            if client.publish(message).is_err() {
-                                Response::error("Can't publish `Get` response")
-                            } else {
-                                Response::ok()
+                    let payload = minimq::publication::DeferedPayload::new(|buf| {
+                        match self.settings.get_json(path, buf) {
+                            Ok(len) => len,
+                            Err(err) => {
+                                // TODO: Lookup failure. Serialize the error as a response.
```

Comment:
I think the best thing to do here is to do a proactive check on the `path` before attempting to defer the payload. If the path doesn't exist, then we bail out early. Open to thoughts on this one

Reply:
The difficulty arises because with the current architecture, we won't realize the path doesn't exist until we go to serialize the publish packet for transmission. I don't know if we can propogate the error all the way back through the stack at that point, so we lose the exact error intel. It could have failed serialization due to a number of other problems, like:
1. No response topic identified
2. Not enough buffer space

Reply:
Don't we always send exactly one packet? I.e.

* start serializing the response packet
* as part of that, try the operation (in the callback, late)
* fail operation for reason Err (absent, not found, buffer too small...) or succeed wit Ok
* try serializing the Err/Ok into the response packet
* if that fails, serialize whatever fixed fatal error message can definitely be serialized
* send the packet

Reply:
;) Well no, we don't. See [below](https://github.com/quartiq/miniconf/pull/179/files#r1302893601) for that discussion.

Reply:
The question is then can we change the topic (from the get topic to the response topic) while serializing if we hit an err?

Reply:
I updated it to properly handle this condition now.